### PR TITLE
dequantize softmax in the TFLite->TOSA pipeline

### PR DESCRIPTION
**Pull request by bjacob**
See https://github.com/google/iree/issues/8974. This is still a 20% end-to-end latency improvement on MobileBert-int8 on configs where matmuls are already reasonably fast, making other things like Softmax more important relatively. That is even after Softmax slowness was much improved recently as observed in #9170. Moreover, discussion around #8974 suggests that the path forward for non-dequantized Softmax is nontrivial, so putting our benchmarks on the dequantized path for now will help insulate them a bit from what we expect to be in-flux for the foreseeable future.

### CODE REVIEW

**Review by rsuderman — APPROVED** [review]

**Comment by bjacob** [comment]
Improvement is as expected:

deq_softmax_improv

https://perf.iree.dev/serie?IREE?MobileBertSquad%20[int8]%20(TFLite)%20big-core,full-inference,experimental-flags%20with%20IREE-Dylib-Sync%20@%20Pixel-6-Pro%20(CPU-ARMv8.2-A)

File: integrations/tensorflow/iree_tf_compiler/TFL/Passes.cpp [hunk 1]
```diff
@@ -52,9 +52,13 @@ void buildTFLImportPassPipeline(OpPassManager &pm) {
   // Convert all TFL ops to TOSA ops
   //----------------------------------------------------------------------------
 
-  mlir::tosa::TOSATFTFLLegalizationPipelineOptions tosaOptions;
   pm.addPass(createLowerGlobalTensorsPass());
+
+  mlir::tosa::TOSATFTFLLegalizationPipelineOptions tosaOptions;
+  // Temporary work-around for https://github.com/google/iree/issues/8974
```
Comment (jpienaar) [inline integrations/tensorflow/iree_tf_compiler/TFL/Passes.cpp:58]: When can we remove this one? Or is this required going forward?
